File: examples/generate_encrypt.rb
require 'active_support'
require 'json'

include ActiveSupport

key_base = 'helloworld'
key_gen  = KeyGenerator.new(key_base, iterations: 1000)

salt      = key_gen.generate_key('test salt')[0, MessageEncryptor.key_len]
sign_salt = key_gen.generate_key('test signed salt')

verifier  = MessageVerifier.new(key_base, serializer: JSON)
encryptor = MessageEncryptor.new(salt, sign_salt, serializer: JSON)

message = { key: 'value' }

puts verifier.generate(message)
puts encryptor.encrypt_and_sign(message)

File: examples/generate_encrypt.rs
extern crate message_verifier;

use message_verifier::{AesHmacEncryptor, DerivedKeyParams, Encryptor, Verifier};

fn main() {
    let key_base = "helloworld";
    let salt = "test salt";
    let sign_salt = "test signed salt";

    let verifier = Verifier::new(key_base);

    let dkp = DerivedKeyParams::default();
    let encryptor = AesHmacEncryptor::new(key_base, salt, sign_salt, dkp).unwrap();

    let message = "{\"key\":\"value\"}";

    println!("{}", verifier.generate(message).expect("Verifier failed"));
    println!(
        "{}",
        encryptor
            .encrypt_and_sign(message)
            .expect("Encryptor failed")
    );
}
